src/command.ts

```diff
@@ -315,7 +315,7 @@ export abstract class Command {
       }
 
       const deprecateAliases = flagDef?.deprecateAliases
-      const aliases = (flagDef?.aliases ?? []).map(a => a.length === 1 ? `-${a}` : `--${a}`)
+      const aliases = ([...flagDef?.aliases ?? [], ...flagDef?.charAliases ?? []]).map(a => a.length === 1 ? `-${a}` : `--${a}`)
```

Comment:
is it possible to put, for example, `c` in both aliases and charAliases such that they'd get the warning multiple times (and we should Set-dedupe them?)

also, rather than iterating all these aliases, should we exit early if `!deprecateAliases` and then compute the aliases and check length?